### ISSUE TITLE
Response model is not displayed correctly

### ISSUE THREAD

**Issue by agliznetsov** [swagger-api/swagger-ui#1690]
I'm running the latest version of swagger-ui from master branch with the following model:

``` json
{
  "swagger": "2.0",
  "paths": {
    "/permissions/{id}": {
      "get": {
        "operationId": "getPermission",
        "parameters": [
          {
            "name": "id",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "schema": {
              "$ref": "#/definitions/Permission"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "Permission": {
      "type": "object",
      "description": "Permission description"
    },
    "UIPermission": {
      "allOf": [
        {
          "$ref": "#/definitions/Permission"
        },
        {
          "type": "object",
          "properties": {
            "actionName": {
              "type": "string"
            }
          },
          "description": "Permission description"
        }
      ]
    },
    "ResourcePermission": {
      "allOf": [
        {
          "$ref": "#/definitions/Permission"
        },
        {
          "type": "object",
          "properties": {
            "resource": {
              "type": "string"
            }
          },
          "description": "Permission description"
        }
      ]
    }
  }
}
```

Note that Permission class has a description and two subtypes.
But the response model is rendered as an empty object: {} without description nor subtypes info.

**Comment by raaga123** [comment]
I am having this problem too. It appears that the response class model schema resolves correctly IF it is defined within the project code base, but not if it is included from an outside library.

Has anybody looked at this?

**Comment by paulwilton** [comment]
yes confirmed having same issue when using external model definition
v2.1.4

**Comment by fehguy** [comment]
@paulwilton please test master for swagger-ui and report back.

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen if you experience otherwise.
